src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -416,22 +417,37 @@ private int recordExitStatus(FilePath workspace, int code) throws IOException, I
         private static final long serialVersionUID = 7599995179651071957L;
         private final OsType os;
         private final String binaryPath;
+        private String architecture;
         private boolean binaryCompatible;
         private boolean binaryCached;
         private boolean cachingAvailable;
 
-        public AgentInfo(OsType os, boolean binaryCompatible, String binaryPath, boolean cachingAvailable) {
+        public AgentInfo(OsType os, String architecture, boolean binaryCompatible, String binaryPath, boolean cachingAvailable) {
             this.os = os;
+            this.architecture = architecture;
             this.binaryPath = binaryPath;
             this.binaryCompatible = binaryCompatible;
             this.binaryCached = false;
             this.cachingAvailable = cachingAvailable;
         }
 
+        // Avoid deserialization issues with newly added fields
+        private Object readResolve() {
+            if (architecture == null || architecture.isEmpty()) {
+                String[] splits = binaryPath.split("_");
+                architecture = splits[splits.length - 1];
+            }
+            return this;
+        }
+
```

Comment:
```suggestion
```

Only relevant for objects serialized persistently, such as to `config.xml` or `program.dat`. This class is only ever sent via Remoting, so we need not worry about version skew.

Reply:
removed in dd9324a8e1e9ed44b845c57cd9d3ddac2ea68939